### DISCUSSION TITLE
Combining shadow and rounded corners

### DISCUSSION THREAD

**Discussion by lars-vc** [davatorium/rofi#1653] [Q&A]
Is there any why to combine shadows and have rounded corners? When using shadows and rounded corners the pointy corner keeps it shadow, this can be fixed by disabling shadows for rofi in for example picom. But is there a way to keep the shadows to that they match the rounded corners?

**Comment by TonCherAmi** [comment]
Try picom's `full-shadow` window-type option, this seems to work for me:

```
wintypes:
{
  normal = { full-shadow = true; };
};
```

**Reply by lars-vc** [reply]
Are you sure that rofi has win type normal? Because this has no effect, xprops gives this as output for rofi:
![image](https://user-images.githubusercontent.com/57136208/177303573-d676a556-ab42-4e74-9148-64a8977ddafc.png)
There is no wintype in there

**Reply by TonCherAmi** [reply]
If I recall correctly I simply tried every single wintype until one worked.

Here's what it looks like with `full-shadow = true`
![image](https://user-images.githubusercontent.com/13941584/177314275-5dc3b4ce-12fd-45e8-b4b0-cf448442c2fa.png)

Here it is with `full-shadow = false`
![image](https://user-images.githubusercontent.com/13941584/177314572-1fdff895-cbcc-4b14-93e6-a3cb4275df83.png)

My `picom.conf` for reference:

<details>

```
backend = "glx";

vsync = true;

fading = true;
fade-delta = 2;
fade-exclude = [
  "class_g != 'Rofi'"
];

shadow = true;
shadow-radius = 18;
shadow-opacity = 0.2;
shadow-exclude = [
  "class_g != 'Rofi'"
];

wintypes:
{
  normal = { full-shadow = true; };
};

```
</details>

**Reply by lars-vc** [reply]
mmh it doesn't seem to work for me, Im just gonna make the rounding of the corners very small so it isn't noticable (sidenote using blur will have the same issue, the corners get a blurry pointy corner). Sad that picom doesn't have a nice fix for all this.
Thanks for the help anyway!

**Comment by marcos-aparicio** [comment]
for anyone still having this issue, when i set wintype = normal as instructed in the previous comment, that affects other settings of mine. So this is what i use to solve this, restricting the full-shadow property only to rofi
```conf
rules = (
  { match = "class_g = 'Rofi'"; full-shadow: true; }
)

```
